### PR TITLE
Extend dataset artifact

### PR DESCRIPTION
**Pull request by yaronha**
extend dataset artifact, support updates, fix doc bug

### CODE REVIEW

File: mlrun/artifacts/base.py [hunk 1]
```diff
@@ -208,3 +214,68 @@ def blob_hash(data):
...
+        if update_spec:
+            artifact_spec.extra_data[prefix + key] = item
+
+
+def get_artifact_meta(artifact):
+    """return artifact object, and list of extra data items
+
+
+    :param artifact:   artifact path (store://..) or DataItem
+    :param stores:     StoreManager object (not required)
```
Comment (Hedingber) [inline mlrun/artifacts/base.py:257]: I don't see it in the args

File: mlrun/artifacts/base.py [hunk 2]
```diff
@@ -208,3 +214,68 @@ def blob_hash(data):
...
+
+    :return artifact object, extra data dict
+
+    """
+    stores = store_manager
```
Comment (Hedingber) [inline mlrun/artifacts/base.py:262]: why this rename ?

File: mlrun/datastore/inmem.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from .base import DataStore, FileStats
+
+
+class InMemStore(DataStore):
```
Comment (Hedingber) [inline mlrun/datastore/inmem.py:18]: Characters don't cost money, suggest to change it to `in_memory.py` file, `InMemoryStore` class, `in_memory_store` variable names etc...

File: mlrun/artifacts/base.py (hunk 1 above)
Comment (yaronha) [reply mlrun/artifacts/base.py]: @Hedingber removed

File: mlrun/artifacts/base.py (hunk 2 above)
Comment (yaronha) [reply mlrun/artifacts/base.py]: legacy, removed

File: mlrun/datastore/inmem.py (hunk 3 above)
Comment (yaronha) [reply mlrun/datastore/inmem.py]: made it longer

**Comment by yaronha** [comment]
@Hedingber fixed and added tests

**Pull request merged**